### PR TITLE
Add config editor and dump viewer

### PR DESCRIPTION
**Pull request by lukeeey**
This PR ports over the config editor and dump viewer from the main site.

We will probably need to set up a redirect for the urls (config_editor -> config-editor), but i'm not sure the best way to do that with docusarus.
one
two
three

### CODE REVIEW

**Comment by Kas-tle** [comment]
I will test this locally later today.

File: docs/other/hydraulic.md [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-
-Any contributions are appreciated. Please feel free to reach out to us on [Discord](https://discord.gg/geysermc) if
-you're interested in helping out with Hydraulic.
-
-### Project Setup {#project-setup}
-
-1. Clone the repo to your computer.
-2. Navigate to the Hydraulic root directory and run `git submodule update --init --recursive`. This command downloads all the needed submodules for Hydraulic and is a crucial step in this process.
-3. The project should import into your IDE after the loom setup is complete. For more detailed information, see the [Fabric setup](https://fabricmc.net/wiki/tutorial:setup)
+---
```
Comment (Kas-tle) [inline docs/other/hydraulic.md:1]: Is this modification intentional?

File: src/pages/utilities/config-editor.tsx [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+        a.dataset.downloadurl = [fileType, a.download, a.href].join(':')
+        a.style.display = 'none'
+        document.body.appendChild(a)
+        a.click()
+        document.body.removeChild(a)
+
+        setTimeout(() => { URL.revokeObjectURL(a.href) }, 1500)
+    }
+
+    function createUUID() {
```
Comment (Kas-tle) [inline src/pages/utilities/config-editor.tsx:235]: is there a reason we can't use `crypto.randomUUID()` ?

Comment (lukeeey) [reply src/pages/utilities/config-editor.tsx:235]: There is not, this was copied over from the previous website code. I’ll change it later

File: i18n/de/code.json [hunk 3]
```diff
@@ -230,6 +230,18 @@
   "pages.download.description.geyseroptionalpack": {
     "message": "An optional Bedrock resource pack to extend Geyser functionality."
   },
+  "pages.configeditor.title": {
+    "message": "Config Editor"
```
Comment (onebeastchris) [inline i18n/de/code.json:234]:
```suggestion
    "message": "Konfigurationseditor"
```

File: i18n/de/code.json [hunk 4]
```diff
@@ -230,6 +230,18 @@
...
+  },
+  "pages.configeditor.subheading": {
+    "message": "Edit your Geyser configuration."
```
Comment (onebeastchris) [inline i18n/de/code.json:237]:
```suggestion
    "message": "Bearbeite deine Geyser Konfigurationsdatei."
```

File: i18n/de/code.json [hunk 5]
```diff
@@ -230,6 +230,18 @@
...
+  },
+  "pages.dumpviewer.title": {
+    "message": "Dump Viewer"
```
Comment (onebeastchris) [inline i18n/de/code.json:240]:
```suggestion
    "message": "Dump Betrachter"
```

File: i18n/de/code.json [hunk 6]
```diff
@@ -230,6 +230,18 @@
...
+  },
+  "pages.dumpviewer.subheading": {
+    "message": "View the breakdown of your Geyser dump."
```
Comment (onebeastchris) [inline i18n/de/code.json:243]:
```suggestion
    "message": "Schau dir eine Zusammenfassung deines Geyser Dumps an."
```

**Review by rtm516 — APPROVED** [review]

**Pull request merged**